pkg/controller/util.go

```diff
@@ -332,6 +332,15 @@ func setAnnotationsFromPodWithPrefix(anno map[string]string, pod *v1.Pod, termMs
 	}
 }
 
+func setLabelsFromTerminationMessage(labels map[string]string, termMsg *common.TerminationMessage) {
+	if labels == nil || termMsg == nil {
+		return
+	}
+	for k, v := range termMsg.Labels {
+		labels[k] = v
+	}
+}
+
```

Comment:
I understand returning if termMsg is nil but why labels?  I think it is a little awkward to always expect that to be allocated.  Maybe do something similar to append builtin?

```golang
func setLabelsFromTerminationMessage(labels map[string]string, termMsg *common.TerminationMessage) map[string]string {
	if ermMsg == nil {
		return labels
	}
        if labels == nil && len(termMsg.Labels) > 0 {
           labels = make(map[string]string)
        }
	for k, v := range termMsg.Labels {
		labels[k] = v
	}
        return labels
}
```

Reply:
I kept it in line with `setAnnotationsFromPodWithPrefix`, which also does not `make` a new map, if the passed in `anno` variable is `nil`. However, it does not even check if `anno` is `nil`. In L413 above the call to `setLabelsFromTerminationMessage` labels are populated, if they are `nil`. IMO it's good practice to still avoid writing to a `nil` map in the func. If you want however, I can add `make` to the func. WDYT?

https://github.com/kubevirt/containerized-data-importer/blob/775c43deead68d7b7106480561ad8fee9b0d8774/pkg/controller/import-controller.go#L413-L421

Reply:
I just think it is weird to have it both ways, either assume labels is not nil and use it without checking, or allocate/return it.  But I think it is better to allocate/return because could be hiding bugs in the other two cases (caller doesn't know have to pass in an allocated map).  However using the map without checking is slightly better because will probably catch the bug in testing

Reply:
Alright, I'll think about it and might open a follow up PR.